test/supports/user_code_import_samples/cls.py

```diff
@@ -14,5 +14,11 @@ def f(self, arg):
     def f2(self, arg):
         return f"other {arg}"
 
+    @modal.method()
+    def calls_f_remote(self, arg):
+        return self.f.remote(arg)
+
+
+UndecoratedC = C  # keep a reference to original class before overwriting
 
-C = app.cls()(UndecoratedC)
+C = app.cls()(C)  # type: ignore[misc]   # "decorator" of C
```

Comment:
Changing this around since the function won't ever import the decorated entity if it has a different name than the underlying one at decoration time